internal/server/iito.go

```diff
@@ -48,56 +48,57 @@ func (s *SOMASServer) runGiftSession() error {
 	return nil
 }
 
-func (s *SOMASServer) getGiftRequests() shared.GiftDict {
-	giftRequestDict := shared.GiftDict{}
+// GetGiftRequests collects all gift requests from the clients in a map
+func (s *SOMASServer) getGiftRequests() map[shared.ClientID]shared.GiftRequestDict {
+	giftRequestDict := map[shared.ClientID]shared.GiftRequestDict{}
 	for id, client := range s.clientMap {
-		giftRequestDict[id] = client.RequestGift()
+		giftRequestDict[id] = client.GetGiftRequests()
 	}
 	return giftRequestDict
 }
-func (s *SOMASServer) getGiftOffers(giftRequestDict shared.GiftDict) (map[shared.ClientID]shared.GiftDict, error) {
-	giftOfferDict := map[shared.ClientID]shared.GiftDict{}
-	var err error
+
+// GetGiftOffers collects all responses from clients to their requests in a map
+func (s *SOMASServer) getGiftOffers(totalRequests map[shared.ClientID]shared.GiftRequestDict) (map[shared.ClientID]shared.GiftOfferDict, error) {
+	totalOffers := map[shared.ClientID]shared.GiftOfferDict{}
+	// Loop over each team
 	for id, client := range s.clientMap {
-		giftOfferDict[id], err = client.OfferGifts(giftRequestDict)
+		// Gather all the requests made to this team
+		requestsToThisTeam := shared.GiftRequestDict{}
+		for fromTeam, indivRequests := range totalRequests {
+			requestsToThisTeam[fromTeam] = indivRequests[id]
+		}
+		offer, err := client.GetGiftOffers(requestsToThisTeam)
 		if err != nil {
-			return giftOfferDict, err
+			totalOffers[id] = offer
 		}
 	}
-	return giftOfferDict, nil
+	return totalOffers, nil
 }
-func (s *SOMASServer) getGiftAcceptance(giftOffersDict map[shared.ClientID]shared.GiftDict) (map[shared.ClientID]shared.GiftInfoDict, error) {
-	acceptedGifts := map[shared.ClientID]shared.GiftInfoDict{}
-	var err error
 
-	receivedByClientDict := make(map[shared.ClientID]shared.GiftDict)
-
-	//puts all the gifts received by a certain client accesible by the id of that client
-	for idsend := range giftOffersDict {
-		for idto := range giftOffersDict {
-			if idsend == idto {
-				continue
-			}
-			if receivedByClientDict[idsend] == nil {
-				receivedByClientDict[idsend] = make(shared.GiftDict)
-			}
-			receivedByClientDict[idsend][idto] = giftOffersDict[idto][idsend]
-		}
-	}
+func (s *SOMASServer) getGiftResponses(totalOffers map[shared.ClientID]shared.GiftOfferDict) (map[shared.ClientID]shared.GiftResponseDict, error) {
+	totalResponses := map[shared.ClientID]shared.GiftResponseDict{}
 
 	for id, client := range s.clientMap {
-		acceptedGifts[id], err = client.AcceptGifts(receivedByClientDict[id])
+		offersToThisTeam := shared.GiftOfferDict{}
+		for fromTeam, indivOffers := range totalOffers {
+			offersToThisTeam[fromTeam] = indivOffers[id]
+		}
+		response, err := client.GetGiftResponses(offersToThisTeam)
 		if err != nil {
-			return acceptedGifts, err
+			totalResponses[id] = response
 		}
```

Comment:
Copied & Pasted statement.
I feel like you got the logic wrong here. This if statement should only be used when an error occurs. Currently the logic is - if there is an error, totalResponses is assigned.   
It might be because it was very late when you did this but lol.

Reply:
yeah will flip the `!=` to `==` I was intending to write tests today, I just wanted to put the PR up first so people could comment bc I did this commit all in 1 go and maps on maps in my head usually do not end well without tests